src/torch_onnx_models/components/_rotary_embedding_utils.py

```diff
@@ -1,77 +1,42 @@
 
+def get_rotary_pos_emb(position_ids: torch.Tensor, cos_cache: torch.Tensor, sin_cache: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
+    """
+    Retrieve the cosine and sine positional embeddings based on the provided position IDs.
 
-def create_rope_caches(
-    config: _configs.ArchitectureConfig,
-) -> tuple[torch.Tensor, torch.Tensor]:
-    # Initialize rope frequencies using the utility function
-
-    inv_freq, attention_factor = _initialize_rope_freqs(config=config)
-
-    # Create position indices for all positions up to max_position_embeddings
-    max_seq_len = config.max_position_embeddings
-    # TODO: position_scale
-    positions = torch.arange(max_seq_len, dtype=torch.float32)
-
-    # Compute the angles for each position and frequency
-    # positions: [max_seq_len], inv_freq: [dim//2] -> angles: [max_seq_len, dim//2]
-    angles = torch.outer(positions, inv_freq)
-
-    # Precompute cos and sin caches
-    cos_cache = torch.cos(angles) * attention_factor
-    sin_cache = torch.sin(angles) * attention_factor
-    return cos_cache, sin_cache
-
-
-def _initialize_rope_freqs(
-    config: _configs.ArchitectureConfig,
-) -> tuple[torch.Tensor, float]:
-    if config.rope_type == "default":
-        return _compute_default_rope_parameters(config)
-    raise ValueError(f"Unsupported rope_type: {config.rope_type}")
-
-
-def _compute_default_rope_parameters(
-    config: _configs.ArchitectureConfig,
-) -> tuple[torch.Tensor, float]:
-    # https://github.com/huggingface/transformers/blob/6dc9ed87a02db8b4ecc26a5e98596cd2bba380b5/src/transformers/modeling_rope_utils.py#L92
-    base = config.rope_theta
-    partial_rotary_factor = config.partial_rotary_factor
-    head_dim = config.head_dim
-    dim = int(head_dim * partial_rotary_factor)
-
-    attention_factor = 1.0  # Unused in this type of RoPE
-
-    # Compute the inverse frequencies
-    inv_freq = 1.0 / (base ** (torch.arange(0, dim, 2, dtype=torch.float) / dim))
-    return inv_freq, attention_factor
+    Args:
+        position_ids (torch.Tensor): The position IDs tensor of shape (batch_size, seq_length).
+        cos_cache (torch.Tensor): The cosine cache tensor of shape (max_position_embeddings, head_dim).
+        sin_cache (torch.Tensor): The sine cache tensor of shape (max_position_embeddings, head_dim).
 
+    Returns:
+        tuple[torch.Tensor, torch.Tensor]: A tuple containing the cosine and sine embeddings,
+                                           each of shape (batch_size, seq_length, head_dim).
+    """
+    return cos_cache[position_ids], sin_cache[position_ids]
 
 # TODO(jambayk): add support for interleaved format if needed
 # requires torch 2.9+
-def apply_rope(
-    x: torch.Tensor,
+# this can actually be fused with get_rotary_pos_emb as well but we keep them separate for clarity
+# as well as to align the model architecture with transformers
+# is it really worth separating this from get_rotary_pos_emb? shape doesn't match transformers
```

Comment:
I think it's reasonable, such that Attention will take QKV not the states with rotary applied?

Reply:
But I don't know better. Maybe sending states in is ok?